src/main/kotlin/com/example/delivery/service/OrderService.kt

```diff
@@ -27,71 +32,96 @@ class OrderService(
 ) {
 
     @LogInvoke
-    fun getById(id: String): DomainOrderWithProduct {
-        return orderRepository.findById(id)?.toDomain()
-            ?: throw NotFoundException("Order with id $id doesn't exists")
+    fun getById(id: String): Mono<DomainOrderWithProduct> {
+        return orderRepository.findById(id)
+            .map { it.toDomain() }
+            .switchIfEmpty(Mono.error(NotFoundException("Order with id $id doesn't exists")))
     }
 
     @SuppressWarnings("ThrowsCount")
-    fun add(createOrderDTO: CreateOrderDTO): DomainOrder {
-        val user = userRepository.findById(createOrderDTO.userId)
-            ?: throw NotFoundException("User with id ${createOrderDTO.userId} doesn't exist")
+    fun add(createOrderDTO: CreateOrderDTO): Mono<DomainOrder> {
+        return userRepository.findById(createOrderDTO.userId)
+            .switchIfEmpty(Mono.error(NotFoundException("User with id ${createOrderDTO.userId} doesn't exist")))
```

Comment:
```suggestion
            .switchIfEmpty {Mono.error(NotFoundException("User with id ${createOrderDTO.userId} doesn't exist")) }
```

no need to create exception each time